src/ucp/core/ucp_rkey.c

```diff
@@ -117,9 +117,6 @@ ucs_status_t ucp_rkey_pack(ucp_context_h context, ucp_mem_h memh,
     UCP_CONTEXT_CHECK_FEATURE_FLAGS(context, UCP_FEATURE_RMA | UCP_FEATURE_AMO,
                                     return UCS_ERR_INVALID_PARAM);
 
-    /* always acquire context lock */
-    UCP_THREAD_CS_ENTER(&context->mt_lock);
```

Comment:
pls review uct_ib_mkey_pack, and other UCTs: they are not really thread safe.

Reply:
They are really not thread safe, but they are per MD / MH. Adding corner test to do parallel rkey_pack for the same MH. The only place I found where it modifies MD is here: https://github.com/openucx/ucx/blob/master/src/uct/ib/mlx5/dv/ib_mlx5dv_md.c#L269

Reply:
but the mkey itself is modified during pack